main.tf

```diff
@@ -49,11 +49,16 @@ module "vpc_endpoints" {
   }
 }
 
+resource "random_id" "unique_prefix" {
+  byte_length = 4
+}
```

Comment:
IIUC, I believe you want to have a [keeper](https://registry.terraform.io/providers/hashicorp/random/latest/docs/resources/id#keepers) set on this like `environment = var.environment`, otherwise the same ID could be used for both (I'm assuming that would be bad).

Reply:
That is a good feature to be aware of. But, not sure is applies to this use case.